src/commonMain/kotlin/exchange.dydx.abacus/state/app/helper/TriggerOrderToastGenerator.kt

```diff
@@ -0,0 +1,380 @@
+        }
+
+        val status = orderPayloadStatus[placeOrderPayload.clientId.toString()]
+            ?: TriggerOrderStatus.Submitting
+        val triggerPrice = formatter.dollar(placeOrderPayload.triggerPrice, tickSize) ?: ""
+
+        val detailStringPath: String?
+        val orderTypePath: String?
+
+        when (placeOrderPayload.type) {
+            "TAKE_PROFIT_MARKET", "TAKE_PROFIT" -> {
+                detailStringPath = when (status) {
+                    TriggerOrderStatus.Submitting -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATING.BODY"
+                    TriggerOrderStatus.Success -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATED.BODY"
+                    TriggerOrderStatus.Failed -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATING_ERROR.BODY"
+                }
+                orderTypePath = when (status) {
+                    TriggerOrderStatus.Submitting -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATING.TITLE"
+                    TriggerOrderStatus.Success -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATED.TITLE"
+                    TriggerOrderStatus.Failed -> "NOTIFICATIONS.TAKE_PROFIT_TRIGGER_CREATING_ERROR.TITLE"
+                }
+            }
+
+            "STOP_LIMIT", "STOP_MARKET" -> {
+                detailStringPath = when (status) {
+                    TriggerOrderStatus.Submitting -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATING.BODY"
+                    TriggerOrderStatus.Success -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATED.BODY"
+                    TriggerOrderStatus.Failed -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATING_ERROR.BODY"
+                }
+                orderTypePath = when (status) {
+                    TriggerOrderStatus.Submitting -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATING.TITLE"
+                    TriggerOrderStatus.Success -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATED.TITLE"
+                    TriggerOrderStatus.Failed -> "NOTIFICATIONS.STOP_LOSS_TRIGGER_CREATING_ERROR.TITLE"
+                }
+            }
+
+            else -> {
+                detailStringPath = null
+                orderTypePath = null
+            }
+        }
+
+        val orderType = orderTypePath?.let { localizer.localize(it) }
+        val detail = detailStringPath?.let {
+            localizer.localizeWithParams(
+                path = it,
+                params = mapOf("NEW_VALUE" to triggerPrice),
+            )
+        }
+
+        return if (orderType != null && detail != null) {
+            Toast(
+                id = placeOrderPayload.clientId.toString(),
+                type = if (status == TriggerOrderStatus.Failed) ToastType.Warning else ToastType.Info,
+                title = orderType,
+                text = detail,
+            )
+        } else {
+            null
+        }
+    }
+
+    private fun showToasts(toasts: List<Toast>) {
+        if (threading == null) {
+            Logger.e { "Threading is null" }
+            return
+        }
+        threading?.async(ThreadingType.main) {
```

Comment:
hmmm it's not clear to me why this is needed, can you explain the thought process? shouldn't need to force this on main thread.

Reply:
To force UI update on main thread.

Reply:
that's unnecessary to enforce inside of abacus. that should be handled by each client.

specifically on ios/android, we transform these callbacks into streams anyways. by default these platforms will consume these streams on the main thread when called from SwiftUI/Compose.

Reply:
This method won't be going through streams.  The client will update the UI directly.

Having Abacus doing the enforcement will simplify the client code.